src/compiler/checker.ts

```diff
@@ -12495,6 +12509,36 @@ namespace ts {
         }
 
         /**
+         * Checks that the return type provided is an instantiation of the global Promise<T> type

```

Comment:
Can you give a quick example where checking assignability instead of exactly matching `Promise<T>` will be a problem, either now or under future downlevelling plans?